src/tiled/terrainbrush.cpp

```diff
@@ -385,16 +378,25 @@ void TerrainBrush::updateBrush(QPoint cursorPos, const QVector<QPoint> *list)
         int leftID = i - 1;
         int rightID = i + 1;
 
-        if((Map::Orientation)mMapOrientation==Map::Staggered){
-            if(y%2==0){
-                bottomID -= 1;
-                leftID -= layerWidth;
-                rightID = rightID + (layerWidth - 1);
-            }
-            else{
-                upperID += 1;
-                leftID = leftID - (layerWidth -1);
-                rightID += layerWidth;
+        QPoint upPoint(p.x(),p.y()-1);
+        QPoint bottomPoint(p.x(),p.y()+1);
+        QPoint leftPoint(p.x()-1,p.y()-1);
+        QPoint rightPoint(p.x()+1,p.y());
+
+        if(mapDocument() && mapDocument()->map() && mapDocument()->map()->orientation() == Map::Staggered){
+            StaggeredRenderer* renderer = dynamic_cast<StaggeredRenderer*>(mapDocument()->renderer());
+            if(renderer){
```

Comment:
Since you're relying on a `dynamic_cast` here, you actually don't need the orientation check above.